config/rbac/leader_election_role.yaml

```diff
@@ -5,26 +5,6 @@ kind: Role
 metadata:
   name: leader-election-role
 rules:
-- apiGroups:
```

Comment:
@enxebre Could/should we do the same RBAC cleanup for control plane and bootstrap provider kubeadm?